lib/utils.pm

```diff
@@ -1960,10 +1960,14 @@ sub install_patterns {
     for my $pt (@pt_list) {
         # if pattern is set default, skip
         next if ($pt =~ /default/);
+        # For Public cloud module test we need skip Instance pattern if outside of public cloud images.
+        next if (($pt =~ /Instance/) && check_var('PUBLIC_CLOUD_IMG', '1'));
```

Comment:
1) I have no idea just match /Instance/ key word is enough or not, should we using more strict match pattern?

2) I think PUBLIC_CLOUD_IMG =1 means you are using public cloud img?

3) think we can check qcow name to decide it is public cloud img or not.  @jlausuch what's your suggestion?

Reply:
I leave it to @asmorodskyi to answer as he is more involved than me in this discussion.

Reply:
1. I think Instance is enough for now, while if you have better solution it is welcome. :)
2. Yes, it is easy to check by the setting. Anyway, if the image name is special and easy to check, I can accept it.

Reply:
I have updated the code to judge public cloud environment through is_public_cloud, please review again. Thanks.